src/js/actions/documents.js

```diff
@@ -680,12 +680,15 @@ define(function (require, exports) {
                 return this.transfer(disposeDocument, document.id, isDocumentSaved);
             }, function () {
                 // the play command fails if the user cancels the close dialog
+            })
+            .then(function () {
+                return this.transfer(toolActions.changeVectorMaskMode, false);
```

Comment:
Does this really need to wait for all the other operations to finish? It seems like it can happen concurrently with everything else and be start immediately with the action.